pkg/trigger/trigger.go

```diff
@@ -117,10 +239,24 @@ func (t *Trigger) waiter() {
 				time.Sleep(delay)
 			}
 
-			if t.params.TriggerFunc != nil {
-				t.params.TriggerFunc()
-			}
+			t.mutex.Lock()
 			t.lastTrigger = time.Now()
+			numFolds := t.numFolds
+			t.numFolds = 0
+			reasons := t.foldedReasons.slice()
```

Comment:
In response to my comment above about whether the folded reasons needs to have its mutex held - I guess since the trigger's mutex is held here, we will be okay.